strider/query_planner.py

```diff
@@ -282,6 +293,10 @@ async def find_valid_permutations(
     for edge in expanded_qg['edges'].values():
         edge['predicate'] = list(edge.pop('filtered_predicates'))
 
+    # Remove edges with no predicates
```

Comment:
When does this happen?

Reply:
This happens on the two lines below

Reply:
Haha, sorry, I mean how would we encounter edges with no predicates? And why do we want to remove them?

Reply:
Good question. This happens because of the filtering step where we remove all predicates where there are no KPs that offer solutions. So if we get a -biolink:regulates->, but we can only solve -biolink:positively_regulates->, we will make that replacement. 

If a predicate is filtered out entirely that means we can't solve this particular edge in this particular direction. That doesn't necessarily mean we have to throw out this query graph, because we might be able to solve the edge in the reverse direction.

Reply:
The reason we have to filter it out is because the permute_qg function doesn't like being passed an empty array to permute (it will return nothing back).